src/main/java/org/prebid/server/bidder/flipp/FlippBidder.java

```diff
@@ -0,0 +1,279 @@
+import org.prebid.server.bidder.flipp.model.response.Decisions;
+import org.prebid.server.bidder.flipp.model.response.Inline;
+import org.prebid.server.bidder.flipp.model.response.Prebid;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderCall;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.flipp.ExtImpFlipp;
+import org.prebid.server.proto.openrtb.ext.request.flipp.ExtImpFlippOptions;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.util.ObjectUtil;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+public class FlippBidder implements Bidder<CampaignRequestBody> {
+
+    private static final String INLINE_DIV_NAME = "inline";
+    private static final Integer COUNT = 1;
+    private static final String CREATIVE_TYPE = "DTX";
+    private static final Set<Integer> AD_TYPES = Set.of(4309, 641);
+    private static final Set<Integer> DTX_TYPES = Set.of(5061);
+    private static final TypeReference<ExtPrebid<?, ExtImpFlipp>> FLIPP_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public FlippBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<CampaignRequestBody>>> makeHttpRequests(BidRequest bidRequest) {
+        final List<HttpRequest<CampaignRequestBody>> httpRequests = new ArrayList<>();
+        final List<BidderError> errors = new ArrayList<>();
+
+        for (Imp imp : bidRequest.getImp()) {
+            final CampaignRequestBody.CampaignRequestBodyBuilder campaignRequestBody = CampaignRequestBody.builder();
+            final ExtImpFlipp extImpFlipp;
+
+            try {
+                extImpFlipp = parseImpExt(imp);
+                campaignRequestBody.ip(resolveIp(bidRequest, extImpFlipp));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+                continue;
+            }
+
+            httpRequests.add(createRequest(bidRequest,
+                    updateCampaignRequestBody(bidRequest, imp, campaignRequestBody, extImpFlipp)));
```

Comment:
Since you are only adding ip to CampaignRequest here, you can move it to `updateCampaignRequestBody` method and not pass around that builder. After that method name will be changed `updateCampaignRequestBody` -> `makeCampaignRequest`

Here is how it will look like: 
```java

private static CampaignRequestBody makeCampaignRequest(BidRequest bidRequest, Imp imp, ExtImpFlipp extImpFlipp) {
        return CampaignRequestBody.builder()
                .ip(resolveIp(bidRequest, extImpFlipp))
                .placements(Collections.singletonList(createPlacement(bidRequest, imp, extImpFlipp)))
                .url(ObjectUtil.getIfNotNull(bidRequest.getSite(), Site::getPage))
                .keywords(resolveKeywords(bidRequest))
                .user(CampaignRequestBodyUser.of(resolveKey(bidRequest, extImpFlipp)))
                .build();
    }
```

Also, after that, `makeHttpRequests` could be simplified:
```java
@Override
    public Result<List<HttpRequest<CampaignRequestBody>>> makeHttpRequests(BidRequest bidRequest) {
        final List<HttpRequest<CampaignRequestBody>> httpRequests = new ArrayList<>();
        final List<BidderError> errors = new ArrayList<>();

        for (Imp imp : bidRequest.getImp()) {
            try {
                final ExtImpFlipp extImpFlipp = parseImpExt(imp);
                final CampaignRequestBody campaignRequest = makeCampaignRequest(bidRequest, imp, extImpFlipp);
                httpRequests.add(makeHttpRequest(bidRequest.getDevice(), campaignRequest));
            } catch (PreBidException e) {
                errors.add(BidderError.badInput(e.getMessage()));
            }
        }

        if (CollectionUtils.isEmpty(httpRequests)) {
            errors.add(BidderError.badInput("Adapter request is empty"));
            return Result.withErrors(errors);
        }

        return Result.of(httpRequests, errors);
    }
```